mailbox/backup/src/main/java/org/apache/james/mailbox/backup/zip/ExtraFieldExtractor.java

```diff
@@ -53,4 +67,29 @@
         }
     }
 
+    public static Optional<SerializedMailboxId> getMailBoxId(ZipEntry entry) throws ZipException {
+        return getStringExtraField(MailboxIdExtraField.ID_AM, entry).map(SerializedMailboxId::new);
+    }
+
+    public static Optional<SerializedMessageId> getMessageId(ZipEntry entry) throws ZipException {
+        return getStringExtraField(MessageIdExtraField.ID_AL, entry).map(SerializedMessageId::new);
+    }
+
+    public static Optional<Long> getSize(ZipEntry entry) throws ZipException {
+        return getLongExtraField(SizeExtraField.ID_AJ, entry);
+    }
+
+    public static Optional<Date> getInternalDate(ZipEntry entry) throws ZipException {
+        ZipExtraField[] extraFields = ExtraFieldUtils.parse(entry.getExtra());
+        return Arrays.stream(extraFields)
+            .filter(field -> field.getHeaderId().equals(InternalDateExtraField.ID_AO))
+            .map(extraField -> ((InternalDateExtraField) extraField).getDateValue())
+            .findFirst()
+            .flatMap(Function.identity());
```

Comment:
I see the statemens:
```java
            .filter(field -> field.getHeaderId().equals(...))
            .map(extraField -> ((....) extraField).getDateValue())
            .findFirst()
            .flatMap(Function.identity());
```
duplicate many times, cany refactor it?

Reply:
nice catch